module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/RenewalsHelperTest.php

```diff
@@ -0,0 +1,433 @@
+<?php
+
+/**
+ * VuFind Helper - Renewals Support Methods
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sambhav Pokharel <sambhavpokharel@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFindTest\ILS\Logic;
+
+use Laminas\Mvc\Plugin\FlashMessenger\FlashMessenger;
+use Laminas\Stdlib\Parameters;
+use PHPUnit\Framework\TestCase;
+use VuFind\ILS\Connection;
+use VuFind\ILS\Logic\RenewalsHelper;
+use VuFind\Validator\CsrfInterface;
+
+use function in_array;
+use function is_array;
+
+/**
+ * Unit tests for the RenewalsHelper
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sambhav Pokharel <sambhavpokharel@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class RenewalsHelperTest extends TestCase
+{
+    /**
+     * Test adding renew details when using an external link.
+     *
+     * @return void
+     */
+    public function testAddRenewDetailsWithLink(): void
+    {
+        $catalog = $this->getMockBuilder(Connection::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['__call'])
+            ->getMock();
```

Comment:
Does this still work if you use the simpler:


```suggestion
        $catalog = $this->createMock(Connection::class);
```

Typically, it should only be necessary to use the mock builder if you want some of the "real" logic to execute -- but in these tests, I don't think we really need to exercise any of the Connection logic; we just need a test double that returns a dummy value. I think we can get the same result mocking the whole class as just mocking the `__call` method. But of course, if I'm wrong about this, it's fine as it is -- just looking to simplify where possible.